src/views/Project/index.jsx

```diff
@@ -0,0 +1,168 @@
+import { Component } from 'react';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import ReactMarkdown from 'react-markdown';
+import { ApolloConsumer } from 'react-apollo';
+import _ from 'lodash';
+import gql from 'graphql-tag';
+import projects from '../../data/loader';
+import BugsTable from '../../components/BugsTable';
+import Issues from './issues.graphql';
+
+class Project extends Component {
+  constructor(props) {
+    super(props);
+    const fileName = props.match.params.projectKey;
+    const projectInfo = projects[fileName];
+    // dictionary with repo as key and tag as value
+    const tmp = projectInfo.repositories
+      ? projectInfo.repositories.reduce(
+          (prev, cur) => ({ ...prev, ...cur }),
+          {}
+        )
+      : {};
+    // initial query search for all repo
+    const repoQuerySearch = ['state:open'];
+    // a dictionary of tag (string) and list of repository in that tag
+    const tagRepoDictionary = Object.keys(tmp).reduce((prev, key) => {
+      const curr = [
+        ...(prev[tmp[key]] || [...repoQuerySearch, `label:${tmp[key]}`]),
+        `repo:${key}`,
+      ];
+
+      return {
+        ...prev,
+        [tmp[key]]: curr,
+      };
+    }, {});
+    const tagRepoList = Object.entries(tagRepoDictionary).map(
+      ([key, value]) => ({
+        label: key,
+        query: value.join(' '),
+      })
+    );
+
+    this.state = {
+      projectInfo,
+      tagRepoList,
+      error: false,
+      loading: true,
+      data: [],
+    };
+  }
+  componentDidMount() {
+    const { tagRepoList } = this.state;
+
+    this.fetchGithubDataNext(tagRepoList);
+  }
+
+  fetchGithubDataNext(tagRepoList) {
+    const { client } = this.props;
+    let allQuery = '';
+
+    if (tagRepoList.length === 0) return;
+
+    tagRepoList.forEach((tag, idx) => {
+      const noCursorQuery = `_${idx}: search(first:100, type:ISSUE, query:"${
```

Comment:
Could we use the decorator approach instead? See my other comment for examples.

Reply:
but how if we dont know the number of query in advance (since for different label, we need to put theme in different query) so it is like this for automation:
```
{_0: search(first:100, type:ISSUE, query:"state:open repo:armenzg/mozilla_ci_tools repo:automatedtester/automation-services-bot repo:automatedtester/powerball-platform repo:automatedtester/unittest-zero repo:davehunt/flynnid repo:davehunt/pytest-mozwebqa repo:mozilla/bidpom repo:mozilla/coversheet repo:mozilla/memchaser repo:mozilla/moz-grid-config repo:mozilla/mozdownload repo:mozilla/mozmill-ci repo:mozilla/nightlytt label:mentored")
      {
      ...Issues
      }
_1: search(first:100, type:ISSUE, query:"state:open repo:marco-c/grcov repo:marco-c/code-coverage-reports repo:marco-c/code-coverage-addon label:good-first-bug")
      {
      ...Issues
      }
_2: search(first:100, type:ISSUE, query:"state:open repo:mozilla-releng/services label:skill:good-first-bug")
      {
      ...Issues
      }
}

fragment Issues on SearchResultItemConnection {
  	issueCount
    nodes{
      ...on Issue  {
        title
        updatedAt
        number
        repository {
          name
        }
        url
        labels(last:2){
          nodes{
            name
            color
          }
        }
        assignees(last:1) {
          nodes{
            login
          }
        }
      }
    }
}
```